### DISCUSSION TITLE
Rules permission for the logged in users

### DISCUSSION THREAD

**Discussion by Salimdevv1** [pocketbase/pocketbase#5098] [Q&A]
i want to make a rule for the users who are logged in so for example the user has access for his emails only when he is logged in and no one else has access to his emails i tried @request.auth.id = id but it doesnt work 
any help please

**Comment by ganigeorgiev** [comment]
You can't control the visibility of individual fields (unless you create a View collection or custom route).

With that said, the access to the `email` field is restricted by default using the `emailVisibility` field and you usually don't have to do anything in addition. The `email` will be returned if one of the below conditions is met:
- is logged in as admin
- is logged in as the owner of the auth record
- `emailVisibility` is explicitly set to true

**Reply by Salimdevv1** [reply]
sorry but i didnt understand , but i just want to make a condition for just the owner of the account has the permission for example to delete his account thats it

**Reply by ganigeorgiev** [reply]
> i just want to make a condition for just the owner of the account has the permission for example to delete his account thats it

Then `@request.auth.id = id` should work.

**Reply by Salimdevv1** [reply]
yeah that's exactly what i did but i doesnt work i dont know why 
![Screenshot from 2024-06-18 16-33-09](https://github.com/pocketbase/pocketbase/assets/130100657/58a9bc8b-8f5d-47d5-a2b8-ae37a7fb80ea)
![image](https://github.com/pocketbase/pocketbase/assets/130100657/8c89312d-3855-4deb-9db2-a5227eec0ef4)

![image](https://github.com/pocketbase/pocketbase/assets/130100657/27039c27-fb5d-449b-88c7-44b794639485)

**Reply by ganigeorgiev** [reply]
I don't understand. Your question was about _" for example to delete his account"_.

If you set `@request.auth.id = id` as Delete API rule this will allow only the authenticated user to delete their account.

If you want the above `getFullList` call to work then make sure that you are authenticated (eg. you could check the auth state with `console.log(pb.authStore.token)` before the call). The JS SDK doesn't know what `$currentUser` is. The auth state is based on what is available in `pb.authStore`.

In any case if you need further assistance please don't treat the discussions as llm chat and try to clearly elaborate what is your use case and what you are trying to do.

**Reply by Salimdevv1** [reply]
"If you set @request.auth.id = id as Delete API rule this will allow only the authenticated user to delete their account."

yeah this is what i want but for the search only the authenticated users can see their emails no one else 
sorry if i didnt explain very well

**Reply by ganigeorgiev** [reply]
Double check that you are authenticated. Again, the JS SDK doesn't know what `$currentUser` is. The auth state is based on what is available in `pb.authStore`.

More than this I don't think I can help further.
